### ISSUE TITLE
How to properly use worker in the browser mode?

### ISSUE THREAD

**Issue by yellow1912** [nextapps-de/flexsearch#251]
Hello,

I'm trying to use this module in the browser. I'm using the following code according to the doc:

```
const { Worker } = require("flexsearch");

var myWorker = new Worker();
```

When I try the search, myWorker.search('some test'); I get the following error:

"TypeError: Cannot read property 'postMessage' of undefined
    at eval (webpack://@somepath./node_modules/.pnpm/flexsearch@0.7.21/node_modules/flexsearch/dist/flexsearch.bundle.js?:23:229)"

I'm using Webpack to build my package, it seems like I missed something obvious but I cannot figure out what.

**Comment by rathboma** [comment]
Having exactly the same issue.

**Comment by rathboma** [comment]
This issue seems to be this line:

https://github.com/nextapps-de/flexsearch/blob/master/src/worker/index.js#L43

I'm running in Electron, and this is incorrectly identifying the process as a node process. It's not.

**Comment by rathboma** [comment]
@ts-thomas Could this line use the following check instead?

```
if (typeof window === 'undefined')
// this is node

```

**Comment by rathboma** [comment]
See code here - https://github.com/flexdinesh/browser-or-node/blob/master/src/index.js

**Comment by ts-thomas** [comment]
Thanks for this hint. I need to solve the merge conflict before I can merge.

**Comment by Gabriel-Alves-Cunha** [comment]
Is this a solved issue?? Cause I've been having the same issue and I'm using version 0.7.21
In the meantime, is there a way to get around this?

**Comment by ts-thomas** [comment]
This fix is included in v0.7.23
